test/server/test_adapter.py

```diff
@@ -541,6 +541,72 @@ def test_query_with_link_model(self):
         }
         assert crn_duns.hash.nunique() == 1000
 
+    def test_match_one_to_many(self, revolution_inc: dict[str, list[str]]):
+        """Test that matching data works when the target has many IDs."""
+        self.setup_database("link")
+
+        crn_x_duns = "deterministic_naive_test.crn_naive_test.duns"
+        crn_wh = self.warehouse_data[0]
+        duns_wh = self.warehouse_data[1]
+
+        res = match(
+            backend=self.backend,
+            source_id=revolution_inc["duns"][0],
+            source=str(duns_wh),
+            target=str(crn_wh),
+            model=crn_x_duns,
+        )
+
+        assert isinstance(res, Match)
+        assert res.source == str(duns_wh)
+        assert res.target == str(crn_wh)
+        assert res.source_id == set(revolution_inc["duns"])
+        assert res.target_id == set(revolution_inc["crn"])
+
+    def test_match_many_to_one(self, revolution_inc: dict[str, list[str]]):
+        """Test that matching data works when the source has more possible IDs."""
```

Comment:
This test looks a bit obscure to me. On line 585 we're asserting that source_id has multiple elements, right?